app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/DisplayDatasetActivity.kt

```diff
@@ -0,0 +1,91 @@
+package com.github.HumanLearning2021.HumanLearningApp.view
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.os.Parcelable
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.BaseAdapter
+import android.widget.GridView
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.lifecycleScope
+import com.github.HumanLearning2021.HumanLearningApp.R
+import com.github.HumanLearning2021.HumanLearningApp.model.CategorizedPicture
+import com.github.HumanLearning2021.HumanLearningApp.model.DummyCategory
+import com.github.HumanLearning2021.HumanLearningApp.presenter.DummyUIPresenter
+import kotlinx.coroutines.launch
+import java.io.Serializable
+
+
+class DisplayDatasetActivity : AppCompatActivity() {
+
+    val fork = DummyCategory("fork")
+    val knife = DummyCategory("knife")
+    val spoon = DummyCategory("spoon")
+
+    val dummyPresenter = DummyUIPresenter()
+
+    val datasetImagesList = ArrayList<CategorizedPicture>()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_display_dataset)
+        lifecycleScope.launch {
+            datasetImagesList.add(dummyPresenter.getPicture(fork.name)!!)
+            datasetImagesList.add(dummyPresenter.getPicture(knife.name)!!)
+            datasetImagesList.add(dummyPresenter.getPicture(spoon.name)!!)
+        }
```

Comment:
There's a race condition there, we don't know when the `.add` chain will finish execution